### PR TITLE
Delay starting node until Genesis time

### PR DESCRIPTION
**Pull request by milosevic**
Refs https://github.com/tendermint/tendermint/issues/2294

* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [x] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: node/node.go [hunk 1]
```diff
@@ -427,6 +429,12 @@ func NewNode(config *cfg.Config,
 
 // OnStart starts the Node. It implements cmn.Service.
 func (n *Node) OnStart() error {
+	now := tmtime.Now()
+	genTime := n.genesisDoc.GenesisTime
+	if genTime.After(now) {
+		time.Sleep(genTime.Sub(now))
```
Comment (melekes) [inline node/node.go]: We should print a message like "Genesis time is in the future. Sleeping until then..."

File: node/node_test.go [hunk 2]
```diff
@@ -75,3 +77,17 @@ func TestSplitAndTrimEmpty(t *testing.T) {
...
 	}
 }
+
+func TestNodeDelayedStop(t *testing.T) {
+	config := cfg.ResetTestRoot("node_delayed_node_test")
+	now := tmtime.Now()
+
+	// create & start node
+	n, err := DefaultNewNode(config, log.TestingLogger())
+	n.GenesisDoc().GenesisTime = now.Add(5 * time.Second)
```
Comment (ValarDragon) [inline node/node_test.go]: Does this reliably pass for smaller values?

Comment (milosevic) [reply node/node_test.go]: It pass for 10 millisecond.

**Comment by ebuchman** [comment]
With this sleep we can't kill the program with SIGINT - see https://github.com/tendermint/tendermint/issues/2434

**Pull request merged**